### ISSUE TITLE
Code Insights: Revisit UX for insight loading state for background data fetching 

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#25170]
## Context

We are going to adopt the Apollo cache system for loading backend insights on the dashboard page. In the nutshell, this system works in this way 

**On the first-page visit** 
1. We open the dashboard page
2. We start loading insights sequentially in parallel by 2 (like we do this now on the sourcegraph.com)
3. While loading is happening we show loading state (insight card with no content but with loader instead)
4. When the backend responds with data we hide the loading state and show either the data or some error state.

The most interesting thing here is that after the backend returned our data we put the in the cache. In the next time when we visit the dashboard page, we could use our cached insight data and show insight immediately without an empty card and loading state at all and load insight data in the background to be sure that we still provide the most relevant data.

So on the second and further renders it works in this way

1. Revisit the dashboard page 
2. See cached insights without loading state 
3. But in the background, we're fetching recent data for insights on the page. **This step requires some UX input**
4. When insights background fetching is over update insights chart if needed. 

## Problem 

We probably need to introduce another loading state to notify users that they're looking at cached data and real data might be different compared with cached (I think it's a corner case but still)

**Comment by AlicjaSuska** [comment]
I've put together a design for this case. We can add the 'Fetching recent data' information below the insights title (see in [Figma](https://www.figma.com/file/sJwUmtrlU9sTQDafh772Gf/Basic-Project-Template-Copy?node-id=246%3A11)).

Copy I've considered:
- 'Updating...' - it's unclear what's going on
- 'Fetching...' - unclear what's being fetched and if the shown data is valid
- 'Fetching data...' - unclear if the shown data is valid
- 'Fetching recent data...' - the most descriptive but a little long

@Joelkw @vovakulikov @felixfbecker Please, share what's your take on the copy is here.

I've also improved the padding and spacing in the chart card to accommodate for this state and avoid 'jumping' of the elements when the fetching is finished. @vovakulikov please, see the [improved card](https://www.figma.com/file/sJwUmtrlU9sTQDafh772Gf/Basic-Project-Template-Copy?node-id=1119%3A2405), I hope all the changes are feasible.

<img width="693" alt="Screenshot 2021-09-22 at 10 07 37" src="https://user-images.githubusercontent.com/20326070/134306664-99c3aaf2-8342-484b-9c26-7c0ad2149a80.png">

**Comment by AlicjaSuska** [comment]
@vovakulikov it's ready for dev, I've marked the issue as 'in progress' and removed myself from asignees. Thank you!

**Comment by Joelkw** [comment]
It seems weird to me @AlicjaSuska @vovakulikov that we want a completely separate loading state for this case rather than trying to reuse our "insights still processing" states. I think this design is fine in isolation but wonder how many different loading states we want to introduce for consistency. Should we just consider throwing back on the "gradient loading message cover" that we have when insights are processing, but change the icon (perhaps make it a spinning loader)? 

I don't think a user needs to know where we're fetching data from (backend or if things are still processing, so call that "fetching from raw data" essentially) and I'd prefer that we keep the states a bit more aligned. What if we just changed the message on our existing full-graph loader?

**Comment by vovakulikov** [comment]
@Joelkw Yeah, I understand that we have a lot of card insight states which is bad cause it is always difficult to remember all of them for users, and don't be confused. But here we probably want to reuse cached data first do not block the insight viewing flow for users while we are updating the cache. 

In other words, I see the point to introduce this state. IMO
